src/views/viewer/CategoryView/CategoryVideos.tsx

```diff
@@ -52,48 +62,65 @@ export const CategoryVideos: React.FC<{ categoryId: string }> = ({ categoryId })
     setIsFiltersOpen((value) => !value)
   }
 
+  const topbarHeight = mdMatch ? 80 : 64
+
   useEffect(() => {
     setVideoWhereInput((value) => ({
       ...value,
       languageId_eq: selectedLanguage,
     }))
   }, [selectedLanguage, setVideoWhereInput])
 
+  useEffect(() => {
+    if (scrollWhenFilterChange) {
+      containerRef.current?.scrollIntoView()
+    }
+
+    // account for videoWhereInput initialization
+    if (!isEqual(videoWhereInput, {})) {
+      setScrollWhenFilterChange(true)
+    }
+    // we dont want to rerun effect when scrollWhenFilterChange changes but only videoWhereInput
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [videoWhereInput])
```

Comment:
If I understand this code correctly, we could just use `ref` instead of `useState`. 
```suggestion
  const scrollWhenFilterChange = useRef(false)
  useEffect(() => {
    if (scrollWhenFilterChange.current) {
      containerRef.current?.scrollIntoView()
    }

    // account for videoWhereInput initialization
    if (!isEqual(videoWhereInput, {})) {
      scrollWhenFilterChange.current = true
    }
  }, [videoWhereInput])
```